erpnext/accounts/doctype/sales_invoice/sales_invoice.js

```diff
@@ -516,6 +516,24 @@ frappe.ui.form.on('Sales Invoice', {
 			};
 		});
 	},
+	//When multiple companies are set up. in case company name is changed set default company address
+	company:function(frm){
+		if (frm.doc.company)
+		{
+			frappe.call({
+				method:"frappe.contacts.doctype.address.address.get_default_address",
+				args:{ doctype:'Company',name:frm.doc.company},
+				callback: function(r){
+					if (r.message){
+						frm.set_value("company_address",r.message)
+					}
+					else {
+						frm.set_value("company_address","")
+					}
+				}
+			})
+		}
+	},
```

Comment:
This will override existing company trigger defined in the [transaction.js](https://github.com/frappe/erpnext/blob/develop/erpnext/public/js/controllers/transaction.js#L375-L403) file. Can you move your code from sales_invoice.js to transaction.js

Reply:
The default company trigger does not seem to be firing on change of the company field. You can check from your end.